app/src/main/java/org/jellyfin/androidtv/ui/WatchedBadge.kt

```diff
@@ -0,0 +1,57 @@
+package org.jellyfin.androidtv.ui
+
+import android.content.Context
+import android.view.LayoutInflater
+import android.view.View
+import android.widget.FrameLayout
+import kotlinx.android.synthetic.main.badge_watched.view.*
+import org.jellyfin.androidtv.R
+import org.jellyfin.androidtv.model.itemtypes.PlayableItem
+
+
+class WatchedBadge(context: Context, val item: PlayableItem? = null) : FrameLayout(context) {
+	private val changeListener: () -> Unit = {
+		item?.let(::setItemData)
+	}
+
+	init {
+		val inflater = LayoutInflater.from(context)
+		inflater.inflate(R.layout.badge_watched, this)
+		item?.let(::setItemData)
+	}
+
+	constructor(context: Context) : this(context, null)
+
+	fun setWatched() {
+		badge_background.visibility = View.VISIBLE
+		badge_watched_tick.visibility = View.VISIBLE
+		badge_counter.visibility = View.GONE
+	}
+
+	fun setRemainingCount(count: Int) {
```

Comment:
I don't think this would happen fast but if the count is higher than 99 we should probably display `99+` so the text won't overflow.